dashboard/app/models/user.rb

```diff
@@ -486,9 +486,8 @@ def email_and_hashed_email_must_be_unique
     # skip the db lookup if we are already invalid
     return unless errors.blank?
 
-    if ((email.present? && (other_user = User.find_by_email_or_hashed_email(email))) ||
-        (hashed_email.present? && (other_user = User.find_by_hashed_email(hashed_email)))) &&
-        other_user != self
+    if (email.present? && self != User.find_by_email_or_hashed_email(email)) ||
+      (hashed_email.present? && self != User.find_by_hashed_email(hashed_email))
```

Comment:
The other code got around it (in an obfuscated way IMO) with `&& other_user != self`, so if other_user was nil the whole equation would evaluate to false.

I like the improved readability you're going for 👍, just need to resolve this error.